src/poetry/core/constraints/version/patterns.py

```diff
@@ -17,7 +17,7 @@
     rf"^~=\s*(?P<version>{VERSION_PATTERN})$", re.VERBOSE | re.IGNORECASE
 )
 X_CONSTRAINT = re.compile(
```

Comment:
Isn't this whole pattern only for `x` in version constraints? Didn't dive too deep into the code, but on quick glance, just changing it still leaves some code related to "x constraint"

Reply:
The code uses the term "x constraint" for some things with `*` in them, but so far as I know this is the only place that is allowing a literal `x`

Reply:
Ah, okay, didn't notice the `*` in the old regex. My bad